fasta/bin.go

```diff
@@ -0,0 +1,142 @@
+package fasta
+
+import (
+	"github.com/vertgenlab/gonomics/dna"
+	"log"
+)
+
+//BinFasta takes in a slice of fastas and breaks it up into x number of fastas with relatively
+//equal sequence in each, where x equals the number of bins specified
+func BinFasta(genome []Fasta, binNum int) map[int][]Fasta {
+	if binNum == 0 {
+		log.Panic("Number of bins must be greater than zero.")
+	}
+
+	var totalBases, lastPos, baseCap, remainder, splitBases, equalBinNum int
+	var lastName string
+	answer := make(map[int][]Fasta, binNum)
+	for c := range genome {
+		totalBases = totalBases + len(genome[c].Seq)
+	}
+
+	//TODO: chance that there's a situation where the final bin with leftOvers is empty or very small, what I have with baseCap,
+	//	or chance that the last bin could be almost double the size of everything else (totalBases/BinNum and remainder goes in last bin)
+	//	could do a check to see if remainder is super large and then handle as necessary
+	equalBinNum = binNum - 1
+	remainder = totalBases % binNum
+	if remainder == 0 {
+		baseCap = totalBases / binNum
+	} else {
+		splitBases = totalBases - remainder //number of bases to split between equalBins
+		baseCap = splitBases / equalBinNum
```

Comment:
Can you help me understand how the bins are "relatively equal"? Judging from baseCap, does this mean each bin has about the same number of bases (as opposed to about the same number of fasta sequences)? Why is "relatively equal" important?

Reply:
the last bin won't be equal to the others, and depending on the math could be much smaller or nearly the same size. The bins other than the last one will all have exactly equal number of bases.